src/kOS.Safe/Compilation/ProgramBuilder.cs

```diff
@@ -99,7 +99,7 @@ protected virtual void AddEndOfProgram(CodePart linkedObject, bool isMainProgram
             else
             {
                 linkedObject.MainCode.Add(new OpcodePush(0)); // all Returns now need a dummy return value on them.
-                linkedObject.MainCode.Add(new OpcodeReturn());
+                linkedObject.MainCode.Add(new OpcodeReturn(0));
```

Comment:
That's correct.  The KSM builder used for COMPILE needs all Opcodes to have a default constructor, so it can make a "dummy" of the opcode during load time to query its attributes (Because C# can't do proper overrides of things without an actual instance present, none of the derived classes of Opcode can have their attributes quered in a static state.  An instance needs to exist in order to check its attributes.  So it has to be able to make a dummy of each just long enough to use Reflection on it and find them, then it throws it away.  That's why they all need a default constructor.

If the default constructor isn't supposed to be used "for real", then make it private so that ONLY the KSM reflection code can use it (which I think is the case for OpcodeReturn.  I made it private so the compiler would help find for me all the places it was being used (like the one above) so I can edit them ).